sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py

```diff
@@ -59,12 +59,12 @@ class DirectoryClient(AsyncStorageAccountHostsMixin, DirectoryClientBase):
     :ivar str location_mode:
         The location mode that the client is currently using. By default
         this will be "primary". Options include "primary" and "secondary".
-    :param str directory_url:
-        The full URI to the directory. This can also be a URL to the storage account
-        or share, in which case the directory and/or share must also be specified.
-    :param share: The share for the directory. If specified, this value will override
+    :param str account_url:
+        The URI to the account. The method `from_directory_url` must be used in order to
```

Comment:
This description needs some updates. The `must be used` sounds like it is imperative that something other than this method is used. It should be softened to something like "In order to create a client given the full URI to the directory, use the `from_directory_url` classmethod."